plugin/rewrite/reverter_test.go

```diff
@@ -0,0 +1,54 @@
+package rewrite
+
+import (
+	"testing"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/pkg/dnstest"
+	"github.com/coredns/coredns/plugin/test"
+
+	"github.com/miekg/dns"
+	"golang.org/x/net/context"
+)
+
+func TestResponseReverter(t *testing.T) {
+	rules := []Rule{}
+	r, _ := newNameRule("stop", "regex", "(core)\\.(dns)\\.(rocks)", "{2}.{1}.{3}", "answer", "name", "(dns)\\.(core)\\.(rocks)", "{2}.{1}.{3}")
+	rules = append(rules, r)
+
+	tests := []struct {
+		from     string
+		fromType uint16
+		answer   []dns.RR
+		to       string
+		toType   uint16
+		noRevert bool
+	}{
+		{"core.dns.rocks", dns.TypeA, []dns.RR{test.A("dns.core.rocks.  5   IN  A   10.0.0.1")}, "core.dns.rocks", dns.TypeA, false},
+		{"core.dns.rocks", dns.TypeSRV, []dns.RR{test.SRV("dns.core.rocks.  5   IN  SRV 0 100 100 srv1.dns.core.rocks.")}, "core.dns.rocks", dns.TypeSRV, false},
+		{"core.dns.rocks", dns.TypeA, []dns.RR{test.A("core.dns.rocks.  5   IN  A   10.0.0.1")}, "dns.core.rocks.", dns.TypeA, true},
```

Comment:
Maybe add a whacky older RR so we test that anything hitting this gets rewritten correctly? I.e a `HINFO` record?

Reply:
>  I.e a HINFO record?

@miekg , had to do some 🏋️ in `plugin/test` 😆